gfx_scene.cpp

```diff
@@ -1364,131 +1381,182 @@ class GfxSceneInternal
+                // AMD Extension support for split metal rough
+                for(uint64_t ext = 0; ext < gltf_material.extensions_count; ++ext)
                 {
-                    material_ref->roughness_map = (*it2).second.first;
-                    material_ref->metallicity_map = (*it2).second.second;
+                    if(strcmp(gltf_material.extensions[ext].name, "AMD_split_roughnessMetallic") == 0)
+                    {
+                        // Need to get the split files out of the extension
+                        std::string json = gltf_material.extensions[ext].data;
+                        json.erase(
+                            remove_if(json.begin(), json.end(),
+                                [](char c) { return (c == '\r' || c == '\t' || c == ' ' || c == '\n'); }),
+                            json.end());
+                        auto get_texture = [&](std::string_view const tag, GfxConstRef<GfxImage> &image) {
+                            if(auto pos = json.find(tag); pos != std::string::npos)
+                            {
+                                pos += tag.length();
+                                std::string clipped = json.substr(pos, json.find_first_not_of("0123456789", pos + 1) - pos);
+                                const auto index = stoi(clipped);
+                                if(index < gltf_model->textures_count)
+                                {
+                                    cgltf_texture const *split_text = &gltf_model->textures[index];
+                                    it = (split_text != nullptr ? textures.find(split_text) : textures.end());
+                                    if(it != textures.end())
+                                    {
+                                        GfxImage *temp = gfxSceneGetObject<GfxImage>(scene, (*it).second);
+                                        temp->format = ConvertImageFormatLinear(temp->format);
+                                        image = (*it).second;
+                                    }
+                                }
+                            }
+                        };
+                        constexpr std::string_view metalTag = "\"metallicTexture\":{\"index\":";
+                        constexpr std::string_view roughTag = "\"roughnessTexture\":{\"index\":";
+                        get_texture(metalTag, material_ref->roughness_map);
+                        get_texture(roughTag, material_ref->metallicity_map);
+                        has_split_extension = true;
+                    }
                 }
-                else
+            }
+            if (!has_split_extension)
+            {
+                cgltf_texture const *metallicity_roughness_map_text = gltf_material_pbr.metallic_roughness_texture.texture;
+                it = (metallicity_roughness_map_text != nullptr ? textures.find(metallicity_roughness_map_text) : textures.end());
+                if(it != textures.end())
```

Comment:
The diff below this point is juts an indent of the existing code to be within the if check. We only search for regular metallicRoughness texture(s) should the extension not be found.